lib/connect.js

```diff
@@ -234,29 +244,39 @@ const getTasks = (model, setupListeners) => {
         setupListeners(mongoClient);
       }
 
-      mongoClient.connect((err, _client) => {
-        ctx(err);
+      /** @type {Promise<never>} */
+      const waitForTunnelError = new Promise(async(_, reject) => {
+        if (tunnel) {
+          const [error] = await once(tunnel, 'error');
+          reject(error);
+        }
+      });
```

Comment:
Hmm... This will resolve immediately in the `race` further in the code and will not allow me to wait for client to connect when there is no tunnel, right? I guess I can remove `if (tunnel)` check and do something like `const [error] = await once(tunnel ?? new EventEmitter(), 'error');` instead so it never resolved, what do you think?

Reply:
Ah, yeah, sorry :) I didn’t realize that this promise was intentionally potentially resolved. Ignore me then :)

Reply:
Reafactored to use never emitting emitter it for the sake of keeeping the code a bit more modern there following your initial suggestion in 6158306 :)